src/mlpack/tests/main_tests/preprocess_one_hot_encode_test.cpp

```diff
@@ -0,0 +1,162 @@
+  dataset = "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;";
+
+  arma::mat matrix;
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;";
+
+  SetInputParam("input", dataset);
+  SetInputParam<vector<int>>("dimensions", {1, 3});
+  mlpackMain();
+
+  arma::mat output = IO::GetParam<arma::mat>("output");
+  REQUIRE(matrix.n_cols == output.n_cols);
+  REQUIRE(matrix.n_rows == output.n_rows);
+  CheckMatrices(output, matrix);
+}
+
+TEST_CASE_METHOD(
+    PreprocessOneHotEncodingTestFixture, "EmptyMatrixTest",
+    "[PreprocessOneHotEncodingMainTest][BindingTests]")
+{
+  arma::mat dataset;
+
+  SetInputParam("input", dataset);
+  SetInputParam<vector<int>>("dimensions", {1, 3});
+  // error since dimesnions are bigger that matrix
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+TEST_CASE_METHOD(
+    PreprocessOneHotEncodingTestFixture, "EmptyIndicesTest",
+    "[PreprocessOneHotEncodingMainTest][BindingTests]")
+{
+  arma::mat dataset;
+  dataset = "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;";
+
+  SetInputParam("input", dataset);
+  SetInputParam<vector<int>>("dimensions", {});
+  // error since dimesnions are bigger that matrix
+  mlpackMain();
+
+  arma::mat output = IO::GetParam<arma::mat>("output");
+  REQUIRE(dataset.n_cols == output.n_cols);
+  REQUIRE(dataset.n_rows == output.n_rows);
+  CheckMatrices(output, dataset);
+}
+
+TEST_CASE_METHOD(
+    PreprocessOneHotEncodingTestFixture, "InvalidDimensionTest",
+    "[PreprocessOneHotEncodingMainTest][BindingTests]")
+{
+  arma::mat dataset;
+  dataset = "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;";
+
+  SetInputParam("input", dataset);
+  SetInputParam<vector<int>>("dimensions", {10000});
+  // error since dimesnions are bigger that matrix
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+TEST_CASE_METHOD(
+    PreprocessOneHotEncodingTestFixture, "NegativeDimensionTest",
+    "[PreprocessOneHotEncodingMainTest][BindingTests]")
+{
+  arma::mat dataset;
+  dataset = "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;"
+            "-1 1 -1 -1 -1 -1 1 -1;"
+            "1 1 -1 -1 -1 -1 1 1;";
+
+  SetInputParam("input", dataset);
+  SetInputParam<vector<int>>("dimensions", {-10000});
+  // error since dimesnions are bigger that matrix
```

Comment:
It looks like this comment got added in a few places; can you make sure it's accurate in each place? :+1: